R/read-policy.R

```diff
@@ -48,3 +48,83 @@ check_keys <- function(policy, verbose = FALSE){
   
   policy
 }
+
+#' Get the Category Names for Policy
+#'
+#' This function identifies the names of all categories in a policy list.
+#'
+#' @param policy R list of a valid policy file
+#' @param verbose Whether or not to return an alert of categories
+#' 
+#' @examples
+#' get_categories(policy_demo, verbose = TRUE)
+#' 
+#' @return A vector of the names of the categories in the policy R list
+#' @importFrom purrr map
+#' @importFrom cli cli_alert_info cli_div cli_text cli_end
+#' @export
+
+get_categories <- function(policy, verbose = FALSE){
+  policy <- flatten_policy(policy)
+  
+  categories <- purrr::map(policy$categories, "category") |>
+    unlist()
+  
+  alert <- function() {
+    cli::cli_div(theme = list(span.emph = list(color = "orange")))
+    cli::cli_text("{.emph Important Message}")
+    cli::cli_end()
+    cli::cli_alert_info("The categories from the policy are {categories}")
+  }
+  
+  if (verbose){
+    alert()
+  }
+  
+  return(categories)
+}
+
+#' Get the Nested Assignments for Category
+#'
+#' This function identifies the nested assignments for a given category.
+#'
+#' @param category a category from a policy R list
+#' @param verbose Whether or not to return an alert of assignments
+#' 
+#' @examples
+#' category <- policy_demo[["categories"]][[1]]
+#' get_nested_assignments(category = category, verbose = TRUE)
+#' 
+#' @return A vector of the names of the nested assignments in the category
+#' @importFrom purrr map
+#' @importFrom cli cli_alert_info cli_div cli_text cli_end
+#' @export
+
+get_nested_assignments <- function(category, verbose = FALSE){
+  name <- category$category
+  assignments <- extract_nested_assignments(category) |> 
+    unlist()
+  alert <- function() {
+    cli::cli_div(theme = list(span.emph = list(color = "orange")))
+    cli::cli_text("{.emph Important Message}")
+    cli::cli_end()
+    cli::cli_alert_info("The assignments from the category {name} are {assignments}")
+  }
+  
+  if (verbose){
+    alert()
+  }
+  
+  return(assignments)
+}
+
+#' @importFrom purrr map
+extract_nested_assignments <- function(category){
```

Comment:
Since I needed to `unlist()` the output, I wrote the recursive bit separately and all the alert messages in `get_nested_assignments`... if you can find a way to write it into the same function, feel free to add it to this branch